src/lib/y2partitioner/actions/add_bcache.rb

```diff
@@ -70,7 +70,9 @@ def usable_blk_devices
         device_graph.blk_devices.select do |dev|
           dev.component_of.empty? &&
             (dev.filesystem.nil? || dev.filesystem.mount_point.nil?) &&
-            (!dev.respond_to?(:partitions) || dev.partitions.empty?)
+            (!dev.respond_to?(:partitions) || dev.partitions.empty?) &&
+            # do not allow nested bcaches, see doc/bcache.md
+            !dev.ancestors.any? { |a| a.is?(:bcache) }
```

Comment:
Wouldn't this be more readable using `#none?` instead of a negated `#any?`

Moreover, what about checking also for CSets to be even safer?

Something like:

```ruby
dev.ancestors.none? { |a| a.is?(:bcache, :bcache_cset) }
```

Reply:
well, none looks good for me. About that CSet it is a bit tricky as Cset is not blocking device and cannot have any children. So ancestor cannot be Cset. But it does not hurt.